File: src/listener.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import String

def callback(data):
    msg = "[Py] Escuche:ros %s" %data.data
    print(msg)
    
rospy.init_node('listener', anonymous=False)
rospy.Subscriber("chatter", String, callback)
rospy.spin()

File: src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"

void chatterCallback(const std_msgs::String::ConstPtr& msg) 
{
  std::cout << "Escuché: " << msg->data.c_str() << std::endl;
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");
  ros::NodeHandle n;
  ros::Subscriber chatter_sub = 
                n.subscribe("chatter", 100, chatterCallback);
  ros::spin();
  return 0;
}
